hello-world/src/lib.rs

```diff
@@ -136,6 +145,45 @@ impl NFTamagotchi for Tamagotchi {
         }
     }
 
+    fn check_attributes(&mut self) {
+        // assert_eq!(msg::source(), program_id(), "Only contract can call this function");
+        
+        let curr_feed_level: u64 = self.calculate_curr_fed();
+        let curr_entertain_level: u64 = self.calculate_curr_entertained();
+        let curr_rest_level: u64 = self.calculate_curr_rest();
+
+        // sending msgs to the owner
+        if curr_feed_level < MIN_ATTRIBUTE {
+            msg::send(
+                self.owner,
+                TmgEvent::FeedMe,
+                0
+            ).expect("Failed to share TmgEvent");
+        }
+        if curr_entertain_level < MIN_ATTRIBUTE {
+            msg::send(
+                self.owner,
+                TmgEvent::PlayWithMe,
+                0
+            ).expect("Failed to share TmgEvent");
+        }
+        if curr_rest_level < MIN_ATTRIBUTE {
+            msg::send(
+                self.owner,
+                TmgEvent::WantToSleep,
+                0
+            ).expect("Failed to share TmgEvent");
+        } 
+
+        // next state check
+        msg::send_delayed(
+            program_id(),
```

Comment:
The more idiomatic way to call a function is to prepend it with module (see https://doc.rust-lang.org/stable/book/ch07-04-bringing-paths-into-scope-with-the-use-keyword.html#creating-idiomatic-use-paths).

```suggestion
            exec::program_id(),
```